backend/plugin/db/pg/pg.go

```diff
@@ -474,16 +474,17 @@ func (driver *Driver) QueryConn(ctx context.Context, conn *sql.Conn, statement s
 	return results, nil
 }
 
-func getStatementWithResultLimit(stmt string, limit int) string {
-	return fmt.Sprintf("WITH result AS (%s) SELECT * FROM result LIMIT %d;", stmt, limit)
-}
-
 func (*Driver) querySingleSQL(ctx context.Context, conn *sql.Conn, singleSQL base.SingleSQL, queryContext *db.QueryContext) (*v1pb.QueryResult, error) {
 	statement := strings.TrimRight(singleSQL.Text, " \n\t;")
 
 	stmt := statement
 	if !strings.HasPrefix(stmt, "EXPLAIN") && queryContext.Limit > 0 {
-		stmt = getStatementWithResultLimit(stmt, queryContext.Limit)
+		var err error
+		stmt, err = getStatementWithResultLimit(stmt, queryContext.Limit)
+		if err != nil {
+			slog.Error("fail to add limit clause due to parser compatibility", log.BBError(err))
```

Comment:
We probably should put "parser compatibility" in the error string as we're not 100% sure. Just state the fact and put error and statement in the log.